### PR TITLE
DPC-1305 get the system from ssas

### PR DESCRIPTION
**Pull request by MrBilnon**
### Fixes [DPC-1305](https://jira.cms.gov/browse/DPC-1305)

Need to return system information for the portal to be able to use, IPs, client tokens and public keys info is returned

### Proposed Changes

* create endpoint to get system from ssas

### Change Details

Does all the same checks as when registering a system for impl/org relationship except this method expects a relationship so asserts that system id is not null and then proxies to ssas

### Security Implications

- [ ] new software dependencies

- [ ] security controls or supporting software altered

- [ ] new data stored or transmitted

- [ ] security checklist is completed for this change

- [ ] requires more information or team discussion to evaluate security implications

- [x] no PHI/PII is affected by this change

### Acceptance Validation

Screen Shot 2021-07-24 at 11 08 05 AM

### CODE REVIEW

File: dpc-go/dpc-api/src/client/ssas_client.go [hunk 1]
```diff
@@ -50,6 +51,25 @@ func NewSsasHTTPClient(config SsasHTTPClientConfig) SsasClient {
...
 
+// GetSystem function to get a ssas system
+func (sc *SsasHTTPClient) GetSystem(ctx context.Context, systemID string) (GetSystemResponse, error) {
+	log := logger.WithContext(ctx)
+
+	url := fmt.Sprintf("%s/%s/%s", sc.config.URL, PostV2SystemEndpoint, systemID)
+
+	resBytes, err := sc.doGet(ctx, url)
+	if err != nil {
+		log.Error("Get ssas system request failed", zap.Error(err))
```
Comment (dhgreene) [inline dpc-go/dpc-api/src/client/ssas_client.go:62]: When we're writing text, let's consistently capitalize acronyms. There are several other ocurrences.
```suggestion
		log.Error("Get SSAS system request failed", zap.Error(err))
```

File: dpc-go/dpc-api/src/v2/ssas_controller_test.go [hunk 2]
```diff
@@ -151,8 +217,23 @@ func (suite *SsasControllerTestSuite) SetupHappyPathMocks() (*http.Request, cont
...
+		GroupID:      "0001",
+		ClientID:     "c001",
+		SoftwareID:   "software-id",
+		ClientName:   "Test Org",
+		APIScope:     "api-scope",
+		XData:        "xdata",
+		LastTokenAt:  "lastToken",
+		PublicKeys:   []map[string]string{{"key": "public-key", "id": "public-key-1", "creation_date": "creation"}},
+		IPs:          []map[string]string{{"ip": "ip", "id": "ip-1", "creation_date": "creation"}},
+		ClientTokens: []map[string]string{{"label": "my-client-token", "id": "public-key-1", "creation_date": "creation", "uuid": "uuid", "expires_at": "expiration"}},
```
Comment (dhgreene) [inline dpc-go/dpc-api/src/v2/ssas_controller_test.go:232]: To reduce the chance of regressions when handling these arrays, should we include another sample in each?

Comment (dhgreene) [reply dpc-go/dpc-api/src/v2/ssas_controller_test.go:232]: As part of our documentation of how we expect the system to work, should we include more realistic examples of test data?

File: dpc-go/dpc-api/src/client/ssas_client.go [hunk 3]
```diff
@@ -50,6 +51,25 @@ func NewSsasHTTPClient(config SsasHTTPClientConfig) SsasClient {
...
+		return GetSystemResponse{}, err
+	}
+	resp := GetSystemResponse{}
+	if err := json.NewDecoder(bytes.NewReader(resBytes)).Decode(&resp); err != nil {
+		log.Error("Failed to convert ssas response bytes to GetSystemResponse model", zap.Error(err))
```
Comment (MrMorie) [inline dpc-go/dpc-api/src/client/ssas_client.go:67]:
```suggestion
		log.Error("Failed to convert SSAS response bytes to GetSystemResponse model", zap.Error(err))
```

**Review by MrMorie — DISMISSED** [review]
LGTM

**Review by MrMorie — APPROVED** [review]
:shipit:

**Pull request merged**